docs/ref/models/indexes.txt

```diff
@@ -0,0 +1,90 @@
+=====================
+Model index reference
+=====================
+
+.. module:: django.db.models.indexes
+   :synopsis: Built-in index types.
+
+.. currentmodule:: django.db.models
+
+Index classes define a specific database index which can be created through
+them. This document explains the API references of :class:`BaseIndex` which is
+the base class for all other index classes. This includes the `index options`_
+and `built-in indexes`_ Django offers. For now the indexes can be created in
+the database via the :class:`~django.db.migrations.operations.CreateIndex` class as follows::
+
+    from django.db import migrations, models
+
+    class Migration(migrations.Migration):
+
+        operations = [
+            migrations.CreateIndex(
+                model_name='Author',
+                models.Index('name', 'age', name='my_index_name') # An index on the name and age fields
+            ),
+        ]
+
+.. note::
+
+    Technically, indexes are defined in `django.db.models.indexes`, but for
+    convenience they're imported into :mod:`django.db.models`; the standard
+    convention is to use ``from django.db import models`` and refer to the indexes
+    as ``models.<Index-name>``.
+
+Index options
+=============
+
+``*fields``
+-----------
+
+.. attribute:: BaseIndex.fields
+
+Index classes take the field names fo which they are being used as arguments
+directly during initialisation.
+
+``model``
+---------
+
+.. attribute:: BaseIndex.model
+
+The model representing the table on which the index has to be set. This is an
+optional argument.
+
+``name``
+--------
+
+.. attribute:: BaseIndex.name
+
+The name of the index, if any you want to keep in the database. If `name` is
+not provided Django will create a name internally on it's own.
+
+Built-in indexes
+================
+
+.. note::
+
+    Technically, these indexes are defined in ``django.db.models.indexes``
+    but for convenience they're available and usually imported from
+    :mod:`django.db.models`.
+
+The various built-in index classes that Django provides are as follows:
+
+``Index``
+---------
+
+.. class:: Index(*fields, **options)
+
+    .. versionadded:: 1.11
```

Comment:
You can put this annotation below the heading at the top of the document rather than repeating it everywhere.